test/pipe.js

```diff
@@ -9,44 +9,44 @@ import {setFixtureDir} from './helpers/fixtures-dir.js';
 
 setFixtureDir();
 
-const pipeToProcess = async (t, fixtureName, funcName) => {
-	const {stdout} = await execa(fixtureName, ['test'], {all: true})[funcName](execa('stdin.js'));
+const pipeToProcess = async (t, index, funcName) => {
+	const {stdout} = await execa('noop-fd.js', [`${index}`, 'test'], {all: true})[funcName](execa('stdin.js'));
 	t.is(stdout, 'test');
 };
 
-test('pipeStdout() can pipe to Execa child processes', pipeToProcess, 'noop.js', 'pipeStdout');
-test('pipeStderr() can pipe to Execa child processes', pipeToProcess, 'noop-err.js', 'pipeStderr');
-test('pipeAll() can pipe stdout to Execa child processes', pipeToProcess, 'noop.js', 'pipeAll');
-test('pipeAll() can pipe stderr to Execa child processes', pipeToProcess, 'noop-err.js', 'pipeAll');
+test('pipeStdout() can pipe to Execa child processes', pipeToProcess, 1, 'pipeStdout');
+test('pipeStderr() can pipe to Execa child processes', pipeToProcess, 2, 'pipeStderr');
+test('pipeAll() can pipe stdout to Execa child processes', pipeToProcess, 1, 'pipeAll');
+test('pipeAll() can pipe stderr to Execa child processes', pipeToProcess, 2, 'pipeAll');
 
-const pipeToStream = async (t, fixtureName, funcName, streamName) => {
+const pipeToStream = async (t, index, funcName) => {
 	const stream = new PassThrough();
-	const result = await execa(fixtureName, ['test'], {all: true})[funcName](stream);
-	t.is(result[streamName], 'test');
-	t.is(await getStream(stream), 'test\n');
+	const result = await execa('noop-fd.js', [`${index}`, 'test'], {all: true})[funcName](stream);
+	t.is(result.stdio[index], 'test');
+	t.is(await getStream(stream), 'test');
 };
 
-test('pipeStdout() can pipe to streams', pipeToStream, 'noop.js', 'pipeStdout', 'stdout');
-test('pipeStderr() can pipe to streams', pipeToStream, 'noop-err.js', 'pipeStderr', 'stderr');
-test('pipeAll() can pipe stdout to streams', pipeToStream, 'noop.js', 'pipeAll', 'stdout');
-test('pipeAll() can pipe stderr to streams', pipeToStream, 'noop-err.js', 'pipeAll', 'stderr');
+test('pipeStdout() can pipe to streams', pipeToStream, 1, 'pipeStdout');
+test('pipeStderr() can pipe to streams', pipeToStream, 2, 'pipeStderr');
+test('pipeAll() can pipe stdout to streams', pipeToStream, 1, 'pipeAll');
+test('pipeAll() can pipe stderr to streams', pipeToStream, 2, 'pipeAll');
 
-const pipeToFile = async (t, fixtureName, funcName, streamName) => {
+const pipeToFile = async (t, index, funcName) => {
 	const file = tempfile();
-	const result = await execa(fixtureName, ['test'], {all: true})[funcName](file);
-	t.is(result[streamName], 'test');
-	t.is(await readFile(file, 'utf8'), 'test\n');
```

Comment:
Many tests are now using `noop-fd.js` instead of `noop.js`. That fixture does not add a newline, which explains why some tests now do not output that newline.